docs/DeveloperGuide.md

```diff
@@ -159,7 +159,39 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Add Modules Feature
+#### Implementation
+
+The add modules mechanism is facilitated by `AddModuleCommand`. Its functionality is implemented in the `AddModuleCommand.java` class which follows the `Command` interface. It extends `Command` with a list of modules `List<Module>` that is to be added to an existing person, as well as the index of the person to add the modules to, stored internally as `modulesToAdd` and `targetIndex` respectively.
+
+Additionally, it implements the following operations: 
+* `Command#execute(Model model)` - Returns the feedback message containing information about module(s) added to a target person, for eventual displays in the GUI.
+* `createEditedPerson(Person personToEdit, List<Module> modulesToAdd)` - Creates and returns a Person with `modulesToAdd` added to the existing details of `personToEdit`.
+* `hasModulesInCommon(Person personToEdit, List<Module> proposedModules)` - Returns true if `personToEdit` contains any modules in `proposedModules`.
+* `getNewModules(Person personToEdit, List<Module> proposedModules)` - Returns List of non-duplicated modules, which may be empty if no unique modules exist.
+* `getCommandResult(...)` - Returns a `CommandResult` with the appropriate feedback depending on if any new modules are added.
+
+Below is a sequence diagram showing the overview of how add modules works:
+![AddModuleSequenceDiagram](images/AddModuleSequenceDiagram.png)
+
+In `AddCommand#execute()`, the internal `createEditedPerson(Person personToEdit, List<Module> modulesToAdd)` is executed. Each `Person` has a `Set<Module>` that represents the Collection of `Modules` associated with that `Person`.
+Hence, we utilize the behaviour of the `Set` data structure to both store and add modules to a person, taking advantage of the fact that Sets contain unique, non-duplicated elements.
 
+Since `Set` is an interface in Java, we instantiate the Set of Modules a person is taking as a `HashSet`.
+
+We then use `HashSet.addAll()` to add the List of New Modules parsed from user input, into the Set of existing Modules. This operation automatically adds any new unique Modules while ignoring Modules that already exist, without requiring any further duplicate-checking on our part.
+
+While this underlying implementation works, we need to capture exactly what new modules were added on top of existing modules, so that we can display a meaningful feedback to our user.
+This can be achieved using the `hasNewModules()` and `getNewModules()` internal helper functions.
+
```

Comment:
Above:
I think you don't need to explain `Set` and its operations, since this is a standard Java class.
The sequence diagram should sufficiently show the execution of the command, so no need to go into so much detail about the execution. We want to focus on explaining the implementation, not the execution of the command. I think it might be better to be less wordy where possible.

Below:
I think it's good that you split into cases to show the possible outcomes.

Reply:
Good point! Will edit this portion to emphasize implementation over execution.